File: data/EXP01/python/2069/2069.output_1.py
# @lc app=leetcode id=2069 lang=python3
class Robot:
    def __init__(self, width: int, height: int):
        self.width = width
        self.height = height
        self.x = 0
        self.y = 0
        self.direction = "East"
    
    def step(self, num: int) -> None:
        for _ in range(num):
            if self.direction == "North":
                if self.y + 1 < self.height:
                    self.y += 1
                else:
                    self.direction = "West"
            elif self.direction == "East":
                if self.x + 1 < self.width:
                    self.x += 1
                else:
                    self.direction = "North"
                    self.y += 1
            elif self.direction == "South":
                if self.y - 1 >= 0:
                    self.y -= 1
                else:
                    self.direction = "East"
                    self.x += 1
            elif self.direction == "West":
                if self.x - 1 >= 0:
                    self.x -= 1
                else:
                    self.direction = "South"
                    self.y -= 1
    
    def getPos(self) -> List[int]:
        return [self.x, self.y]
    
    def getDir(self) -> str:
        return self.direction

File: data/EXP01/golang_no_system_var/2069/2069.output_1.go
type Robot struct {
	width, height   int
	x, y            int
	direction       string
}

func NewRobot(width int, height int) *Robot {
	return &Robot{
		width:     width,
		height:    height,
		x:         0,
		y:         0,
		direction: "East",
	}
}

func (r *Robot) Step(num int) {
	for num > 0 {
		switch r.direction {
		case "North":
			r.y++
			if r.y >= r.height {
				r.y--
				r.direction = "West"
			}
		case "East":
			r.x++
			if r.x >= r.width {
				r.x--
				r.direction = "North"
			}
		case "South":
			r.y--
			if r.y < 0 {
				r.y++
				r.direction = "East"
			}
		case "West":
			r.x--
			if r.x < 0 {
				r.x++
				r.direction = "South"
			}
		}
		num--
	}
}

func (r *Robot) GetPos() []int {
	return []int{r.x, r.y}
}

func (r *Robot) GetDir() string {
	return r.direction
}


